text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+
+Hotfixes for packages can be releases as-needed, and there is no limit to the number of patch releases in a given `major.minor.patch` range.
+
+## Component packages
+
+In theory, we could support component packages for every framework implementation. As a result, the following packages would be siblings of each other:
+
+```
+packages
+├── button
+├── button-angular
+├── button-react
+└── button-vue
+```
+
+However, we could limit these packages to only core packages, if needed. We could also move components to their own sub-packages folder, for example `packages/react` or a top-level `react` set of packages.
+
+The primary component package would be found under `packages/button` and would contain the following contents:
+
+- The Vanilla.js implementation of the component, this is to facilitate sharing with frameworks
+- The style representation of the component, initially this would be Sass but could in theory be multiple formats that are offered
+- A specification for the component (where applicable)
+- Metadata around the component (could be a generated artifact) to be consumed by websites to aid in search result ranking
+- Tests for the component
+- A development "story" for storybook
+- Other exports that could be useful for sites
+  - Usage content for the component (mdx)
+  - Component docs (mdx)
+
+The package name itself would follow the convention `@carbon/<component>`, for example `@carbon/button`. Each component is given it's own package to allow for the following situations:
+
+- Ability to see usage information around a specific component
+- Allow add-on teams to import components as needed (see [Community-contributed components](#community-contributed-components))
+- Allow ability to use individual components as-needed
+- Allow framework-specific solutions to easily add in components and re-export them as-needed
+
+### Frameworks
+
+A framework-specific component package would be found under `packages/button-<framework>`. For example, `packages/button-react`. This package would import `@carbon/button` (found at `packages/button`) and has the option to use the Vanilla.js implementation if applicable.
+
+The framework-specific package would include:
+
+- The framework implementation of the component, using Vanilla as needed
+- A specification for the component
+- Metadata for the component
+- Tests
+- A development "story" for the component
+- Imports styles from `@carbon/button` and re-exports them
+
+We include **both** styles and implementation so that instead of having to do:
+
+```bash
+yarn add @carbon/button @carbon/button-react
+```
+
+One can write:
+
+```bash
+yarn add @carbon/button-react
+```
+
+As it would contain everything needed to use the component, while still relying on the core package and receiving updates.
+
+In addition, each framework would have an entrypoint for core components. For React, this package would be found under: `packages/react` and would be published to `@carbon/react`.
```

Comment:
Yup! Exactly. Essentially `@carbon/react` re-exports all core component packages by doing:

```
export { Accordion, AccordionItem } from '@carbon/button-accordion';
export { default as Button } from '@carbon/button-react';
// ...
```

Reply:
This also enables add-ons to implement the same contract as core where they only bring in the components they need instead of having to subscribe to all of core (although this is just an idea at this stage).